docker/client.py

```diff
@@ -140,9 +140,14 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if self._auth_configs:
                 if headers is None:
                     headers = {}
-                headers['X-Registry-Config'] = auth.encode_full_header(
-                    self._auth_configs
-                )
+                if utils.compare_version('1.19', self._version) >= 0:
```

Comment:
From Josh's answer, seems like the change only affects 1.7.0 and was reverted in 1.7.1 - making this PR unnecessary. Users of 1.7.0 should upgrade to 1.7.1.

@moutten Can you confirm that this change is unnecessary with Docker 1.7.1 ?